protocol-specs/generators/transaction_merkle_root_for_blocks/index.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const {
+	bufferToHex,
+	hash,
+	intToBuffer,
+	getRandomBytes,
+} = require('@liskhq/lisk-cryptography');
+const BaseGenerator = require('../base_generator');
+
+const getRandomTransactionIds = count => {
+	const ids = [];
+	for (let index = 0; index < count; index += 1) {
+		ids.push(getRandomBytes(32));
+	}
+	return ids;
+};
+
+const LEAFPREFIX = intToBuffer(0);
+const BRANCHPREFIX = intToBuffer(1);
+
+const concat = (m1, m2) => Buffer.concat([m1, m2]);
+
+const leafHash = m => hash(concat(LEAFPREFIX, m));
+
+const branchHash = m => hash(concat(BRANCHPREFIX, m));
+
+const merkleRoot = transactionIds => {
+	const len = transactionIds.length;
+	if (len === 0) return Buffer.from([]);
```

Comment:
should return emptyHash
```suggestion
	if (len === 0) return hash(Buffer.from([]));
```